modules/ossm-config-openshift-monitoring-kiali.adoc

```diff
@@ -0,0 +1,88 @@
+//
+// * service-mesh-docs-main/observability/ossm-kiali-assembly.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="ossm-config-openshift-monitoring-kiali_{context}"]
+= Configuring OpenShift Monitoring with Kiali
+:context: ossm-config-openshift-monitoring-kiali
+
+//Need some kind of intro
+//Possible file name may change
+//Possible assembly file may change
+//Assemblies, topic map info needs to be worked out still for 3.0, now that docs has a better idea of what is to be included for 3.0 TP1.
+
+//FIX THE STYLE THINGS ONCE DEV HAS REVIEWED/APPROVED THE CONTENT. Content came from different places, so the first priority is ensuring the steps are in the correct order, and are accurate for 3.0 TP1.
+
+The following steps show how to integrate {KialiProduct} with `user-workload monitoring`.
+
+.Prerequisites
+
+* {SMProductName} is installed.
+
+* `user-workload monitoring` is enabled. See "Enabling user-defined projects".
+
+* {KialiProduct} 1.86 is installed.
+
+.Procedure
+
+. Create a `ClusterRoleBinding` for Kiali:
++
+.Example `ClusterRoleBinding` configuration
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kiali-monitoring-rbac
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+subjects:
+- kind: ServiceAccount
+  name: kiali-service-account
+  namespace: istio-system
+----
+
+. Find out the revision name of your Istio instance:
++
+.Example configuration
+[source,terminal]
+----
+$ kubectl get istiorevisions.sailoperator.io
+NAME   READY   STATUS    IN USE   VERSION   AGE
+test   True    Healthy   True     v1.21.0   119m9091
+----
+
+. Create a Kiali resource and point it to your Istio instance:
++
+.Example configuration
+[source,yaml]
+----
+apiVersion: kiali.io/v1alpha1
+kind: Kiali
+metadata:
+  name: kiali-user-workload-monitoring
+  namespace: istio-system
+spec:
+  external_services:
+    istio:
+      config_map_name: istio-test # <1>
+      istio_sidecar_injector_config_map_name: istio-sidecar-injector-test # <2>
+      istiod_deployment_name: istiod-test # <3>
+      url_service_version: 'http://istiod-test.istio-system:15014/version' #<4>
+    prometheus:
+      auth:
+        type: bearer
+        use_kiali_token: true
+      thanos_proxy:
+        enabled: true
+      url: https://thanos-querier.openshift-monitoring.svc.cluster.local:9091
+----
+<1> Replace `istio-test` with your revision name.
+<2> Replace `istio-sidecar-injector-test` with your revision name.
+<3> Replace `istiod-test` with your revision name.
+<4> Replace 'http://istiod-test.istio-system:15014/version` with your revision name.
+
+// The community docs from which this came and has been taken, https://github.com/istio-ecosystem/sail-operator/tree/main/docs#integrating-with-kiali, do not have validation steps. It is possible there are not validation steps for TP1. Question out to dev 08/21/2024, waiting a response.
```

Comment:
It actually would be good to add a validation step here! Something that should be added to the community docs as well. The validation step should be

1. Getting the Kiali URL from the `Route`:
    ```
    echo "https://$(kubectl get routes -n istio-system kiali -o jsonpath='{.spec.host}')"
    ```
2. Opening Kiali in your browser by following the URL.